src/app/pages/credentials/certificates-dash/forms/acmedns-form/acmedns-form.component.ts

```diff
@@ -0,0 +1,149 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component } from '@angular/core';
+import { FormGroup, Validators } from '@angular/forms';
+import { FormBuilder, FormControl } from '@ngneat/reactive-forms';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { TranslateService } from '@ngx-translate/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { DnsAuthenticatorType } from 'app/enums/dns-authenticator-type.enum';
+import { helptextSystemAcme as helptext } from 'app/helptext/system/acme';
+import { DnsAuthenticator } from 'app/interfaces/dns-authenticator.interface';
+import { DynamicFormSchema } from 'app/interfaces/dynamic-form-schema.interface';
+import { FormErrorHandlerService } from 'app/modules/ix-forms/services/form-error-handler.service';
+import { WebSocketService } from 'app/services';
+import { IxSlideInService } from 'app/services/ix-slide-in.service';
+
+interface DnsAuthenticatorList {
+  key: string;
+  variables: string[];
+}
+
+@UntilDestroy()
+@Component({
+  templateUrl: './acmedns-form.component.html',
+  styleUrls: ['./acmedns-form.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AcmednsFormComponent {
+  get isNew(): boolean {
+    return !this.editingAcmedns;
+  }
+
+  get title(): string {
+    return this.isNew
+      ? this.translate.instant(helptext.add_title)
+      : this.translate.instant(helptext.edit_title);
+  }
+
+  form = this.formBuilder.group({
+    name: [null as string, Validators.required],
+    authenticator: [DnsAuthenticatorType.Cloudflare, Validators.required],
+    attributes: this.formBuilder.group({}),
```

Comment:
This is untyped unfortunately. Can we type this?

Reply:
Would you please clarify, to which property this comment is related. Is it the value of `attributes`?

Reply:
We use formbuilder from ngneat/reactive-forms because it provides typing. It infers types from the first element, so here `name` is a string, etc. Attributes will not have any types, because of `{}`, types would need to be specified explicitly `attributes: this.formBuilder.group<Something>({})`.

Reply:
I've implement this and other adjustments in my next PR https://github.com/truenas/webui/pull/6650 , please see commit "https://github.com/truenas/webui/pull/6650"